esp-hal/src/sdm.rs

```diff
@@ -0,0 +1,257 @@
+//! Sigma-Delta modulation peripheral driver.
+
+use fugit::HertzU32;
+
+use crate::{
+    clock::Clocks,
+    gpio::{OutputSignal, PeripheralOutput},
+    peripheral::{Peripheral, PeripheralRef},
+    peripherals,
+    private,
+};
+
+/// Sigma-Delta modulation peripheral driver.
+pub struct Sdm<'d, SD> {
```

Comment:
Drop the generic from this, please. If there is only a single instance in any of the devices, it's just overcomplication for no reason.